draft-ietf-quic-tls.md

```diff
@@ -724,6 +703,35 @@ This results in abandoning loss recovery state for the Initial encryption level
 and ignoring any outstanding Initial packets.
 
 
+### Discarding Handshake Keys
+
+An endpoint MUST NOT discard its handshake keys until the TLS handshake is
+confirmed ({{handshake-confirmed}}).  An endpoint SHOULD discard its handshake
+keys as soon as it has confirmed the handshake.  Most applications protocols
+will send data after the handshake, generating acknowledgements and ensuring
+that both endpoints can discard their handshake keys promptly.  Endpoints that
+do not have reason to send immediately after completing the handshake MAY send
+ack-eliciting frames, such as PING, which will cause the handshake to be
+confirmed when they are acknowledged.
+
```

Comment:
Why not make this a MUST, or at least SHOULD, such that a hanshake timeout can close much earlier than Idle timeout?

Reply:
There's no real reason to discard those keys, so it's enough to rely on other incentives.

Reply:
I agree with @martinthomson, discarding handshake keys is not a security requirement. And in practice if an application protocol never sends ACK-soliciting frames it's probably not very useful...